cmd/oauth.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/manifoldco/manifold-cli/analytics"
+	"github.com/manifoldco/manifold-cli/api"
+	"github.com/manifoldco/manifold-cli/config"
+	"github.com/manifoldco/manifold-cli/middleware"
+	"github.com/urfave/cli"
+)
+
+var (
+	linkedOK = "Your Manifold account is now linked"
+)
+
+func init() {
+	oauthCmd := cli.Command{
+		Name:     "oauth",
+		Usage:    "Change your account to authenticate with an OAuth provider",
+		Category: "AUTHENTICATION",
+		Subcommands: []cli.Command{
+			{
+				Name:   "github",
+				Usage:  "Link to a GitHub account using the OAuth 2 Web Flow",
+				Action: linkGitHubWeb,
+			},
+			{
+				Name:      "github-token",
```

Comment:
I think there should probably only be a top level command per-provider and use the flags like you originally had to differentiate between the approaches to authentication